internal/command/jsonstate/state.go

```diff
@@ -495,6 +495,13 @@ func marshalResources(resources map[string]*states.Resource, module addrs.Module
 	return ret, nil
 }
 
+// SetSensitive marks the SensitiveValues in Resource as nil to display the sensitive values.
+// This function is only used when the -show-sensitive flag is passed as an argument in
+// tofu state show command.
+func (r *Resource) SetSensitive() {
```

Comment:
The name is a little bit confusing, maybe we should call it `SetNonsensitive()` instead?

Reply:
agreed, how about `SetSensitiveAsNil()`?

Reply:
`SetSensitiveAsNil()` can work, but from reading it, the first thing that pops into my head is that we took a specific sensitive value and replaced it with nil value instead.
Maybe `MakeAllValuesNonsensitive`?

Reply:
sounds good, thanks

Reply:
I have removed this method as I think it's no longer needed and we can reset the sensitive value in [state_show.go](https://github.com/opentofu/opentofu/blob/94cd210f21d448024bd6d63183390a1e683f7548/internal/command/state_show.go#L176) itself.